### PR TITLE
feat (provider/cohere): Use Cohere v2 API.

### LINKED ISSUE #3331: Support Cohere v2 API
### Feature Description

Cohere recently released [v2 of their API ](https://cohere.com/blog/new-api-v2), which includes support to define tools in JSON schema instead of Python types.

### Use Case

Supporting v2 enables complex tool definitions, which are now limited to primitive types, as mentioned in the docs https://sdk.vercel.ai/providers/ai-sdk-providers/cohere#model-capabilities. 

### Additional context

https://docs.cohere.com/v2/docs/migrating-v1-to-v2

### PR DESCRIPTION
**Pull request by shaper**
Work items:
- [x] streaming
- [x] tool use
- [x] review tests
- [x] update docs

Resolves #3331

### CODE REVIEW

File: packages/cohere/src/cohere-chat-language-model.ts
Comment (lgrammel) [inline packages/cohere/src/cohere-chat-language-model.ts:211]: Each tool call needs an ID, ideally the one sent by the provider API. If the provider does not send IDs for tool calls, we generate them.

File: packages/cohere/src/cohere-prepare-tools.ts
Comment (lgrammel) [inline packages/cohere/src/cohere-prepare-tools.ts:35]: Some tool choices can be simulated by e.g. removing tools. So `auto` and `none` are always possible. For `required` or `tool`, you need a way to force the LLM to make a tool call.

Comment (shaper) [reply packages/cohere/src/cohere-prepare-tools.ts]: I've revised to implement `auto` and `none`. I don't see a way to force a call in Cohere, so I've made `required` throw. `tool` follows what Mistral does, which appears to be:
- filter out all tools other than the one specified
- leave it to the LLM (unless Mistral actually supports `any`?) to decide whether to call it (so it's not a forced call)

File: packages/cohere/src/cohere-chat-language-model.ts
Comment (shaper) [reply packages/cohere/src/cohere-chat-language-model.ts]: Revised to use provider's tool id.

**Comment by shaper** [comment]
PTAL. I am still looking at the `cohere-chatbot-with-tools.ts` example which breaks due to `tool_calls` in an assistant message in what appears to be a schema-incompatible spot, but if there is a fix needed there we can land it in a separate PR IMO.

File: packages/cohere/src/cohere-chat-language-model.test.ts
Comment (shaper) [inline packages/cohere/src/cohere-chat-language-model.test.ts:350]: Woke up and remembered this -- will work on un-skipping these, should probably fix before merge.

**Review by lgrammel — APPROVED** [review]
pls update / activate stream tests

**Pull request merged**
